contracts/Strategy.sol

```diff
@@ -11,36 +11,138 @@ import {
+
+    function transfer(
+        IERC20 token,
+        address from,
+        address to,
+        uint256 share
+    ) external;
+
+    function deposit(
+        IERC20 token,
+        address from,
+        address to,
+        uint256 amount,
+        uint256 share
+    ) external payable returns (uint256 amountOut, uint256 shareOut);
+
+    function setMasterContractApproval(
+        address user,
+        address masterContract,
+        bool approved,
+        uint8 v,
+        bytes32 r,
+        bytes32 s
+    ) external;
+
+    function toAmount(
+        IERC20 token,
+        uint256 share,
+        bool roundUp
+    ) external view returns (uint256 amount);
+
+    function toShare(
+        IERC20 token,
+        uint256 amount,
+        bool roundUp
+    ) external view returns (uint256 share);
+
+    function totals(IERC20) external view returns (Rebase memory totals_);
+
+    function withdraw(
+        IERC20 token_,
+        address from,
+        address to,
+        uint256 amount,
+        uint256 share
+    ) external returns (uint256 amountOut, uint256 shareOut);
+}
+
+interface IAbracadabra {
+    function COLLATERIZATION_RATE() external view returns (uint256);
+
+    function bentoBox() external view returns (IBentoBoxV1);
+
+    function masterContract() external view returns (address);
+
+    function addCollateral(
+        address to,
+        bool skim,
+        uint256 share
+    ) external;
+
+    function removeCollateral(address to, uint256 share) external;
+
+    function borrow(address to, uint256 amount)
+        external
+        returns (uint256 part, uint256 share);
+
+    function repay(
+        address to,
+        bool skim,
+        uint256 part
+    ) external returns (uint256 amount);
+
+    function exchangeRate() external view returns (uint256);
+
+    function userBorrowPart(address) external view returns (uint256);
+
+    function userCollateralShare(address) external view returns (uint256);
+
+    function totalBorrow() external view returns (Rebase memory totals);
+
+    function collateral() external view returns (address);
+
+    function accrue() external;
+}
+
+
 contract Strategy is BaseStrategy {
     using SafeERC20 for IERC20;
     using Address for address;
     using SafeMath for uint256;
 
-    // Units used in Maker contracts
-    uint256 internal constant WAD = 10**18;
-    uint256 internal constant RAY = 10**27;
 
-    // DAI token
+    // Collateral Rate Precision
+    uint256 internal constant C_RATE_PRECISION = 1e5;
+
+    // Exchange Rate Precision
+    uint256 internal constant EXCHANGE_RATE_PRECISION = 1e18;
+
+    // MIM token
     IERC20 internal constant investmentToken =
-        IERC20(0x6B175474E89094C44Da98b954EedeAC495271d0F);
+        IERC20(0x99D8a9C45b2ecA8864373A26D1459e3Dff1e17F3);
 
     // 100%
-    uint256 internal constant MAX_BPS = WAD;
+    uint256 internal constant MAX_BPS = 100;
```

Comment:
@newmickymousse seems like MAX_BPS is used to calculate debt repayment. You probably want to be able to handle more precision than 1% of the total capital allocated.

Reply:
yes, you are right.